Tree resolver

cybersonic opened the pull request:
# Description

This is a massive change, 
1. adding cftokens to parse tests
2. Speeding up the tests so they run in chunks
3. Output for each test matches commadbox testbox output
4. Added better highlighting of test results within the editor. 
5. Added a resolver so changes run updates and we dont have to do a full scan 
6. Speeded up the running of the folder scan (again with a resolver so tests are lazy loded until run in the tree) 

## Type of change

Please delete options that are not relevant.

- [ ] Bug Fix
- [ ] Improvement
- [ ] New Feature

Commit:
Finalized changelog for v3.0.0

mark-distrokid committed:
WIP

mark-distrokid committed:
Improving the accuracy of parsing of tests by adding cftokens

mark-distrokid committed:
Updating ESLint. still issues with included libs running tests

mark-distrokid committed:
Working on codecoverage from lucee execution logs, more parsing of tokens, mxunit compat and moving to Typescript since that will give better documentation

mark-distrokid committed:
Moving things to the resources

mark-distrokid committed:
Keeping in sync

mark-distrokid committed:
Finishing up the speed up and onChange updates

mark-distrokid committed:
Updating the speed of tree rendering

mark-distrokid committed:
Speeding up running of the test. REmoved breaking queue manager (implemetned simple queue)

mark-distrokid committed:
Adding Change log

mark-distrokid committed:
Include node_modules in release package
Updates package version and removes unused dependencies to ensure that required node_modules are packaged. This change fixes distribution issues by bundling only the necessary modules.

lmajano commented:
Thanks, will start reviews soon. I see lots of conflicts though?

cybersonic commented:
I did update from master. Will do another sync to make sure there are no
issues.

Mark Drew


On Thu, 29 May 2025 at 23:04, Luis Majano ***@***.***> wrote:

> *lmajano* left a comment (Ortus-Solutions/vscode-testbox#20)
> <https://github.com/Ortus-Solutions/vscode-testbox/pull/20#issuecomment-2920687739>
>
> Thanks, will start reviews soon. I see lots of conflicts though?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ortus-Solutions/vscode-testbox/pull/20#issuecomment-2920687739>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABEGHFOMB4BXI6UX7GSPJD3A575PAVCNFSM6AAAAAB6FVT2UGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMRQGY4DONZTHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

mark-distrokid committed:
Fixing conflicts with remote master
Merge branch 'master' into tree_resolver

mark-distrokid committed:
Tiny fixes and adding boxlang and commandbox extensions as dependencies so it doesnt break

mark-distrokid committed:
Formating as well as fixing assigning the results to tests with duplicate names

cybersonic commented:
@lmajano I have removed the conflicts but do have questions about boxlang. It looks like it's now a requirement to have the boxlang extension installed as otherwise you get a 
`Unknown language in contributes.vscode-testbox.language. Provided value: boxlang`

lmajano commented:
Let me investigate. Unfortunately this might be a requirement since I was starting to use the LSP and parsers.  I am not sure I can remove the dependency anymore. 

I will have to make a bundle for this.  Give me some time

cybersonic commented:
| have been investigating and there might be a way to register boxlang in the testbox extension but just limited functionality (just enable the test runner) until you get it. I will test now

mark-distrokid committed:
Registering the boxlang language in case you dont have it.

This removes the dependency on an extension you might not be using.

cybersonic commented:
I have a version that works. IT defines the boxlang language in this extension, but no contributions (aside from testing) and doesnt then need any dependencies

mark-distrokid committed:
Formatting and linting fixes

mark-distrokid committed:
Linter Fixes and adding docs

mark-distrokid committed:
Linting, Fixing tests and removing console.log so that Jest runs like a dream

mark-distrokid committed:
Adding testing of the Jest unit tests

mark-distrokid committed:
Running in Jest

mark-distrokid committed:
Removing the tests runner in github.

There is some kind of finding file issue that would need to debug

cybersonic commented:
@lmajano will this PR be blocked until you add the LSP? Also would mean I have to remove existing parser?

lmajano commented:
I think we can do phases. To not stagnate progress. Give me some time for reviews. And we can push forward.

cybersonic commented:
Ping @lmajano

lmajano commented:
Pong! 🏓

lmajano commented on src/utils/resultRenderer.js:
```diff
@@ -0,0 +1,203 @@
+function renderBundleOrSuiteResult( bunbdleOrSpec, run, tablevel = 0 ) {
+	run.appendOutput( `${RESET}` );
+
+	const titleprefix = prefix[bunbdleOrSpec.status || prefix.Passed];
```
@cybersonic

lmajano commented:
src/utils/resultParser.js:61

errorMessage is never set before being passed to run.errored, resulting in an empty message. Replace with the actual failure message, e.g. specResult.failMessage.
run.errored( test, errorMessage, specResult.totalDuration );

src/utils/luceeExecLogParser.ts:7

There's a typo in the class name (Exection vs. Execution). Rename to LuceeExecutionReport (and likewise for LuceeExectionLog and LuceeExectionMetric).
export class LuceeExectionReport {

src/utils/resultRenderer.js:74

The string interpolation is missing a $ around results.CFMLEngineVersion, so it prints {results.CFMLEngineVersion} literally. Change to ${results.CFMLEngineVersion}.
run.appendOutput( `${BOLD}CFML Engine\t${results.CFMLEngine} v {results.CFMLEngineVersion}${nl}` );

package.json:5

The version was downgraded from 3.0.0 to 2.0.3, which violates semantic versioning. It should be bumped forward (e.g., 3.x.x) rather than decreased.
"version": "2.0.3",

cybersonic committed:
Merge branch 'Ortus-Solutions:development' into tree_resolver

cybersonic committed:
#20 Pull Request review changes.

cybersonic commented:
> src/utils/resultParser.js:61
> 
> errorMessage is never set before being passed to run.errored, resulting in an empty message. Replace with the actual failure message, e.g. specResult.failMessage. run.errored( test, errorMessage, specResult.totalDuration );

It is set, though?
```
errorMessage = new vscode.TestMessage( specResult.failMessage );
...
run.errored( test, errorMessage, specResult.totalDuration );
break;
```

> 
> src/utils/luceeExecLogParser.ts:7
> 
> There's a typo in the class name (Exection vs. Execution). Rename to LuceeExecutionReport (and likewise for LuceeExectionLog and LuceeExectionMetric). export class LuceeExectionReport {
>
This has been removed.  It wasn't attached and could be its own extension. No need to litter testbox with this just yet


> src/utils/resultRenderer.js:74
> 
> The string interpolation is missing a $ around results.CFMLEngineVersion, so it prints {results.CFMLEngineVersion} literally. Change to ${results.CFMLEngineVersion}. run.appendOutput( `${BOLD}CFML Engine\t${results.CFMLEngine} v {results.CFMLEngineVersion}${nl}` );
> 
Fixed 

> package.json:5
> 
> The version was downgraded from 3.0.0 to 2.0.3, which violates semantic versioning. It should be bumped forward (e.g., 3.x.x) rather than decreased. "version": "2.0.3",
Updated to 3.0.1

lmajano commented:
Can we now try to do incremental PRs so they are not so huge :)
This way, they become easier to track for potential regressions.
I appreciate all the hard work you are doing on this amigo!

cybersonic commented:
Will do absolutely. This was a big performance change, so it was a big chunk. Hopefully, I will keep them neat and tidy next time! :) 

I appreciate the feedback and the work you are doing!
